File: server/models/textbook.py
from pydantic import BaseModel
from typing import List, Optional


class Section(BaseModel):
    title: str
    page: int


class Chapter(BaseModel):
    number: str
    title: str
    page: int
    sections: List[Section] = []


class TextbookMetadata(BaseModel):
    title: str
    s3_key: str = ""          # kept for backward compat; maps to local path
    local_key: str = ""
    user_id: str
    document_type: str
    table_of_contents: List[Chapter] = []

File: client-next/types/textbook.ts
export interface Section {
  title: string;
  page: number;
}

export interface Chapter {
  number: string;
  title: string;
  page: number;
  sections: Section[];
}

export interface TextbookMetadata {
  title: string;
  s3_key: string;
  local_key?: string;
  user_id: string;
  document_type: string;
  table_of_contents: Chapter[];
}

export interface BookItem {
  title: string;
  s3_key: string;
  file_id?: string;
  author?: string;
  num_pages?: number;
}

export interface ChapterNode {
  id: string;
  title: string;
  sections: SectionNode[];
}

export interface SectionNode {
  id: string;
  title: string;
}

export interface BookStructure {
  chapters: ChapterNode[];
}
